ppa/archive/management/commands/generate_textcorpus.py

```diff
@@ -191,10 +28,36 @@ def add_arguments(self, parser):
             "The default of -1 considers ALL documents.",
         )
 
+    def iter_solr(self, nsize=10, item_type='page'):
+        i=0
+        q=SolrQuerySet().search(item_type=item_type)
+        total = q.count()
+        for i in progressbar(range(0, total, nsize)):
+            q.set_limits(i,i+nsize)
+            yield from q
+
+    def iter_pages(self):
+        for d in self.iter_solr(item_type='page'):
+            yield {k:v for k,v in d.items() if k in self.PAGE_OUTPUT_FIELDS}
+
+    def iter_works(self):
+        for d in self.iter_solr(item_type='work'):
+            yield d
 
     def handle(self, *args, **options):
-        with logging_disabled():
-            SolrCorpus(
-                path=options["path"],
-                doc_limit=options["doc_limit"],
-            ).save()
\ No newline at end of file
+        path = options['path']
+        os.makedirs(path, exist_ok=True)
+        path_meta = os.path.join(path,'metadata.jsonl')
+        path_texts = os.path.join(path,'pages.jsonl')
+        
+        with jsonlines.open(path_meta,'w') as of_meta:
+            for i,d in enumerate(self.iter_works()):
+                of_meta.write(d)
+                if options['doc_limit']>0 and i+1>=options['doc_limit']: 
+                    break
+        
+        with jsonlines.open(path_texts,'w') as of_meta:
+            for i,d in enumerate(self.iter_pages()):
+                of_meta.write(d)
+                if options['doc_limit']>0 and i+1>=options['doc_limit']: 
+                    break
```

Comment:
why is the enumeration and doc limit filtering needed?

would likely be more efficient to iterate and save in chunks

Reply:
jsonlines by default saves by row, I'm not aware of chunking. The doclimit param is a little silly, legacy of previous script, I don't see why anyone would want a partial export where for the gensim script it made more sense, maybe we can just can it.

Reply:
If we need a doclimit, I think it should be applied at the solr query level - maybe a page limit would make more sense for the new version.  How long does the script take? Any value in a shorter run for dev/test? If not I agree about dropping the option.